### PR TITLE
DAOS-11527 tests: Check return value in cart test

### PR DESCRIPTION
**Pull request by jolivier23**
CID 105861: Check return value of pthread_create

Required-githooks: true

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: src/tests/ftest/cart/threaded_server.c [hunk 1]
```diff
@@ -107,29 +107,39 @@ int main(int argc, char **argv)
...
-		pthread_create(&thread[rc], NULL, progress, &status);
+	if (rc != 0) {
+		printf("Failed to create context: " DF_RC "\n", DP_RC(rc));
+		goto fini;
+	}
+
+	for (thread_count = 0; thread_count < NUM_THREADS; thread_count++) {
+		rc = pthread_create(&thread[thread_count], NULL, progress, &status);
+		if (rc != 0) {
+			printf("Failed to create thread: %d, exiting\n", rc);
```
Comment (daltonbohning) [inline src/tests/ftest/cart/threaded_server.c:136]: Do we want to format the error code?
```suggestion
			printf("Failed to create thread:  %d (%s), exiting\n", rc, strerror(rc));
```

File: src/tests/ftest/cart/threaded_server.c [hunk 2]
```diff
@@ -140,11 +150,13 @@ int main(int argc, char **argv)
...
 
+	rc = 0;
+stop:
 	status = STOP;
 	printf("Waiting for threads to stop\n");
 
-	for (rc = 0; rc < NUM_THREADS; rc++)
-		pthread_join(thread[rc], NULL);
+	for (i = 0; i < thread_count; i++)
+		pthread_join(thread[i], NULL);
```
Comment (daltonbohning) [inline src/tests/ftest/cart/threaded_server.c:159]: Do we need to check the return value of `pthread_join`?

File: src/tests/ftest/cart/threaded_server.c (hunk 1 above)
Comment (jolivier23) [reply src/tests/ftest/cart/threaded_server.c]: Fixed

File: src/tests/ftest/cart/threaded_server.c (hunk 2 above)
Comment (jolivier23) [reply src/tests/ftest/cart/threaded_server.c:160]: coverity doesn't complain about it so I'm going to punt it til later.  There isn't much we would do with that error.

**Review by daltonbohning — APPROVED** [review]

**Review by soumagne — APPROVED** [review]

**Pull request merged**